src/lib/common/include/sol-types.h

```diff
@@ -249,6 +249,28 @@ struct sol_blob *sol_blob_ref(struct sol_blob *blob);
 void sol_blob_unref(struct sol_blob *blob);
 void sol_blob_set_parent(struct sol_blob *blob, struct sol_blob *parent);
 
+enum sol_key_value_type {
+    SOL_KEY_VALUE_TYPE_STRING,
+    SOL_KEY_VALUE_TYPE_INT,
+    SOL_KEY_VALUE_TYPE_INVALID
```

Comment:
Better move `INVALID` to be the first one, so checking if a `sol_key_value` is valid is cheaper (comparison with zero is cheaper than with any other number).